pkg/trie/triedb/codec/node.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package codec
+
+const ChildrenCapacity = 16
+
+// MerkleValue is a helper enum to differentiate between inline and hashed nodes
+// https://spec.polkadot.network/chap-state#defn-merkle-value
+type MerkleValue interface {
+	isMerkleValue()
+	IsHashed() bool
+}
+
+type (
+	// Value bytes as stored in a trie node
+	InlineNode struct {
+		Data []byte
+	}
+	// Containing a hash used to lookup in db for real value
+	HashedNode struct {
+		Data []byte
+	}
```

Comment:
Do these need to be exported?

Reply:
Actually no, I'll change it! thks

Reply:
My bad, I was checking and yeah, since we are using them in `TrieDB` we need to export them